src/tracer.js

```diff
@@ -231,13 +231,24 @@ export default class Tracer {
             ctx.traceId = parent.traceId;
             ctx.spanId = Utils.getRandom64();
             ctx.parentId = parent.spanId;
-            ctx.flags = parent.flags;
-
             // reuse parent's baggage as we'll never change it
             ctx.baggage = parent.baggage;
 
-            parent.finalizeSampling();
-            ctx.finalizeSampling();
+            // Parent exists without a sampling decision
+            // An example of where this happens is in mobile clients where sampler parameters
+            // are not propagated
+            if ((parent.flags & constants.DEFERRED_SAMPLING_MASK)
+                === constants.DEFERRED_SAMPLING_MASK) {
+                ctx.flags = parent.flags ^ constants.DEFERRED_SAMPLING_MASK
```

Comment:
I think we need to remove the deferred_sampling_mask in the parent.flag too and also set the sampled_mask on it too if it should be sampled. This is because the same parent context can be used to create multiple spans. We don't want to run this logic multiple times, only once on the first span using this parent span. Future spans will just respect the parent context which has been modified correctly.